python/file_generator.py

```diff
@@ -0,0 +1,377 @@
+			ts_schema=self.ts_schema_df, cbsa_data=self.cbsas)
+			#tracts=tracts, counties=counties) #instantiate edits rules engine
+
+		#Loads LAR and TS data to the rules engine. 
+		rules_check.load_lar_data(lar_data)
+		rules_check.load_ts_data(ts_data)
+
+		#Runs the edits against the LAR row and produces edit check results. 
+		for func in dir(rules_check):
+			if func[:1] in ("s", "v") and func[1:4].isdigit()==True:
+				#print("applying:", func)
+				getattr(rules_check, func)()
+		
+		#Returns edit check results. 
+		return rules_check.results
+
+	def make_ts_row(self):
+		"""
+		Uses the lar_gen object to create a TS row, 
+		dictionary object.
+		"""
+		
+		self.ts_row = self.lar_gen.make_ts_row(self.data_map)
+
+		return self.ts_row
+
+	def make_clean_lar_row(self, ts_row):
+		"""Uses the lar_gen object and a TS row to create a LAR row that 
+		passes syntax and validity edits according to the FIG."""
+
+		#Stores the stop condition and the initial number of iterations. 
+		stop = False
+		iters = 1
+
+		#Makes a new row using the lar generator. 
+		row = self.lar_gen.make_row(lei=self.lei)
+		
+		#Begins a loop that creates the LAR row. The loop generates the row 
+		#with the lar_generator and then validates the row
+		#against the rules engine for syntax or validity edits. 
+
+		#If syntax or validity edits are present, the row is run through 
+		#the contraints and validated again. 
+
+		while stop == False:
+
+			#Copies row to enable diff logging.
+			row_base = row.copy() 
+			
+			#Creates an edit report based on the validation.
+			res = pd.DataFrame(self.validation(row, ts_row))
+			
+			#Logs the results of edits that have failed. 
+			logging.info(res[res.status=="failed"]) 
+			
+			#If there are no syntax or validity edits present, the stop
+			#condition is invoked and the row is returned. 
+			if len(res[res.status=="failed"])<=0:
+				stop = True 
+			
+			#If there are syntax or validity edits present, the constraints
+			#are applied and revalidated while stop is False. 
+			else:
+				message = "\nstarting constraints iteration {iter}".format(
+					iter=iters)
+				logging.info(message)
+				row = self.constraints_loop(self.get_const_list(), row, 
+					row_base) 
+			
+			iters+=1
+
+		return row
+		
+
+	def create_files(self, kind):
+		"""Creates a clean file or a set of edit files specified by the
+		function call"""
+
+		#Creates TS row data. 
+		self.ts_row = self.make_ts_row()
+
+		#Creates a TS row dataframe. 
+		ts_df = pd.DataFrame(self.ts_row, index=[1])
+
+		#The following produces a clean file. 
+		if kind == 'clean_file':
+
+			#Creates a first row and a second row of LAR data which result in 
+			#a LAR dataframe for which more rows are added. 
+			for i in range(0, self.file_length):
+				print('Creating row {i}'.format(i=i))
+				if i==0:
+					lar_row = self.make_clean_lar_row(ts_row=self.ts_row)
+					first_row = pd.DataFrame(lar_row, index=[1])
```

Comment:
What's the reason that the first two rows need to be created separately?

Reply:
Can we have more description about the process of creating the file in the comments?

Reply:
Yes, I just added more description. I also reduced the file creation process so that it creates a first row separately from all of the other rows.